### PR TITLE
 Eval to validate lastest expectation for filterings by sources and topics

### PR DESCRIPTION
**Pull request by YuriHechter**

### CODE REVIEW

File: search/quality_eval.go [hunk 1]
```diff
@@ -245,10 +258,15 @@ type HitSource struct {
 // https://archive.kbb1.com/he/events/meals
 // https://archive.kbb1.com/he/events/friends-gatherings
 // https://archive.kbb1.com/he/events?year=2013
-func ParseExpectation(e string) Expectation {
+func ParseExpectation(e string, db *sql.DB) Expectation {
+	originalE := e
 	if e == "" {
 		return Expectation{ET_EMPTY, "", nil, e}
 	}
+	takeLatest := strings.HasPrefix(e, PREFIX_LATEST)
```
Comment (bbfsdev) [inline search/quality_eval.go]: Lower "e" so that [Latest] will also work well.

File: search/quality_eval.go [hunk 2]
```diff
@@ -539,3 +569,83 @@ func WriteToCsv(path string, records [][]string) error {
 	}
 	return nil
 }
+
+func getLatestUidByFilters(filters []Filter, db *sql.DB) (string, error) {
```
Comment (bbfsdev) [inline search/quality_eval.go]: Nice!

Please test this function, it is fragile, i.e., expected to change with time.

Super! Now after you get UID it is as if it was a standard expectation, at some section, right?

File: search/quality_eval.go (hunk 1 above)
Comment (YuriHechter) [reply search/quality_eval.go]: done

File: search/quality_eval.go (hunk 2 above)
Comment (YuriHechter) [reply search/quality_eval.go]: Tests added

File: search/quality_eval_test.go [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+func (suite *QualityEvalSuite) TearDownSuite() {
+	// Close connections.
+	common.Shutdown()
+	// Drop test database.
+	suite.Require().Nil(suite.DestroyTestDB())
+}
+
+// In order for 'go test' to run this suite, we need to create
+// a normal test function and pass our suite to suite.Run
+func TestEngine(t *testing.T) {
```
Comment (bbfsdev) [inline search/quality_eval_test.go]: s/TestEngine/TestEval/

File: search/quality_eval_test.go [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+	r.Nil(err)
+	r.Equal(latestUID, resultUID)
+
+	fmt.Printf("Test by source and topic \n")
+	resultUID, err = search.GetLatestUidByFilters([]search.Filter{sourceFilter, tagsFilter}, common.DB)
+	r.Nil(err)
+	r.Equal(latestUID, resultUID)
+}
+
+func (suite *QualityEvalSuite) updateSourceParent(child mdbmodels.Source, parentSource mdbmodels.Source, insertChild bool, insertParent bool) error {
```
Comment (bbfsdev) [inline search/quality_eval_test.go]: Don't we have those functions already here:
https://github.com/Bnei-Baruch/archive-backend/blob/master/es/indexer_test.go#L273

What is new that we use only here?
Maybe we extract those to common lib?
WDYT?

Comment (bbfsdev) [reply search/quality_eval_test.go]: I mean also the two functions below.

File: search/quality_eval_test.go [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+
+	childTag := mdbmodels.Tag{Pattern: null.String{"ibur", true}, ID: 1, UID: "L2jMWyce"}
+	childSource := mdbmodels.Source{Pattern: null.String{"bs-akdama-zohar", true}, ID: 1, TypeID: 1, UID: "ALlyoveA"}
+	parentTag := mdbmodels.Tag{Pattern: null.String{"arvut", true}, ID: 2, UID: "L3jMWyce"}
+	parentSource := mdbmodels.Source{Pattern: null.String{"bs-akdmot", true}, ID: 2, TypeID: 1, UID: "1vCj4qN9"}
+
+	r.Nil(suite.updateTagParent(childTag, parentTag, true, true))
+	r.Nil(suite.updateSourceParent(childSource, parentSource, true, true))
+
+	for i := 1; i < 13; i++ {
```
Comment (bbfsdev) [inline search/quality_eval_test.go]: Why 13?

Comment (bbfsdev) [reply search/quality_eval_test.go]: What this loop do?

File: search/quality_eval.go [hunk 6]
```diff
@@ -539,3 +569,83 @@ func WriteToCsv(path string, records [][]string) error {
 	}
 	return nil
 }
+
+func GetLatestUidByFilters(filters []Filter, db *sql.DB) (string, error) {
```
Comment (bbfsdev) [inline search/quality_eval.go]: What [latest] expectations we support here?
Please add comment with description like in parse expectation function, see comment in lines 249 to 260.

File: search/quality_eval_test.go [hunk 7]
```diff
@@ -0,0 +1,315 @@
...
+	if err != nil {
+		return err
+	}
+
+	// Drop test DB
+	_, err = db.Exec("DROP DATABASE " + m.testDB)
+	return err
+}
+
+func (m *TestDBManager) runMigrations(testDB *sql.DB) error {
```
Comment (bbfsdev) [inline search/quality_eval_test.go]: Nice, don't we already have this in index_test?
Can we reuse code?

**Pull request merged**